src/game/server/tf/tf_player.cpp

```diff
@@ -11464,19 +11464,62 @@ void CTFPlayer::CheckSpellHalloweenDeathGhosts( const CTakeDamageInfo &info, CTF
 	if ( TF_IsHolidayActive( kHoliday_HalloweenOrFullMoon ) )
 	{
 		int iHalloweenDeathGhosts = 0;
-		CTFWeaponBase *pWeapon = dynamic_cast<CTFWeaponBase *>( info.GetWeapon() );
+		CTFWeaponBase* pWeapon = dynamic_cast<CTFWeaponBase *>( info.GetWeapon() );
+		
+		if ( IsPlayerClass( TF_CLASS_ENGINEER ) )
+		{
+			CObjectSentrygun* pSentry = dynamic_cast<CObjectSentrygun*>( info.GetInflictor() );
+			CTFProjectile_SentryRocket* pSentryRockets = dynamic_cast<CTFProjectile_SentryRocket*>( info.GetInflictor() );
 
-		// was this a wrangler kill?
-		if ( info.GetDamageCustom() == TF_DMG_CUSTOM_PLAYER_SENTRY )
+			if ( ( info.GetDamageCustom() == TF_DMG_CUSTOM_PLAYER_SENTRY ) && iHalloweenDeathGhosts == 0 ) // Was the Wrangler used + spelled?
+			{
+				CTFLaserPointer* pLaserPointer = dynamic_cast<CTFLaserPointer*>( GetEntityForLoadoutSlot( LOADOUT_POSITION_SECONDARY ) );
+				if ( pLaserPointer )
+				{
+					CALL_ATTRIB_HOOK_INT_ON_OTHER(pLaserPointer, iHalloweenDeathGhosts, halloween_death_ghosts);
+				}
+			}
+			if ( ( pSentry || pSentryRockets ) && iHalloweenDeathGhosts == 0 ) // Was the Sentry used & the Wrench is spelled?
```

Comment:
You may be able to mirror what halloween_pumpkin_explosions is doing and just check the attribute on the owner, not necessarily on the wrench itself.